### PR TITLE
Fetchart: unify source handling, split out of #1917

### PR DESCRIPTION
**Pull request by wordofglass**
As suggested by @sampsyo in #1917, this is my Fetchart refactoring in a separate PR. It adds no functionality, but may come in handy if I actually get around implementing said PR.

Opened for now to run the tests.

### CODE REVIEW

File: beetsplug/fetchart.py [hunk 1]
```diff
@@ -41,9 +41,80 @@
 CONTENT_TYPES = ('image/jpeg', 'image/png')
 DOWNLOAD_EXTENSION = '.jpg'
 
-CANDIDATE_BAD = 0
-CANDIDATE_EXACT = 1
-CANDIDATE_DOWNSCALE = 2
+
+class Candidate():
```
Comment (sampsyo) [inline beetsplug/fetchart.py]: Should probably inherit from object to get a new-style class.

**Comment by sampsyo** [comment]
Looking good so far! It also looks like the tests need work. Thank you for updating the docs too!

File: beetsplug/fetchart.py (hunk 1 above)
Comment (wordofglass) [inline beetsplug/fetchart.py]: :white_check_mark:

File: test/test_art.py [hunk 2]
```diff
@@ -367,12 +367,7 @@ def test_leave_original_file_in_place(self):
         self.assertExists(self.art_file)
 
     def test_delete_original_file(self):
-        config['import']['delete'] = True
-        self._fetch_art(True)
-        self.assertNotExists(self.art_file)
-
-    def test_move_original_file(self):
-        config['import']['move'] = True
+        self.plugin.src_removed = True
```
Comment (wordofglass) [inline test/test_art.py]: This obviously takes some detail out of the test. It is however consistent with the way `cautious` is tested, which isn't changed in the config either in `FSArtTest`.
Moving the config reads to FetchArtPlugin__init__ seemed cleaner to me, it makes keeping this test as-is difficult, though

**Comment by sampsyo** [comment]
Sorry for forgetting about this PR! The refactoring is great; I like the way the `Candidate` class turned out. Thanks for doing the hard work! :sparkles:

**Comment by wordofglass** [comment]
Well, I'm mostly scratching my own itches as I'm moving my library to beets :smile:

**Pull request merged**